### PR TITLE
Fix a memory leak in `OctreeBase::operator=`

### PR DESCRIPTION
**Pull request by kunaltyagi**
Reported by: @maciejmatuszak

---
A better approach would be `unique_ptr` but grepping for `root_node_` shows that it'd not be a simple task

### CODE REVIEW

File: octree/include/pcl/octree/octree_base.h [hunk 1]
```diff
@@ -247,6 +247,10 @@ class OctreeBase {
   {
     leaf_count_ = source.leaf_count_;
     branch_count_ = source.branch_count_;
+    if (root_node_)
+    {
+        delete root_node_;
+    }
```
Comment (kunaltyagi) [inline octree/include/pcl/octree/octree_base.h]:
```suggestion
    if (root_node_) {
      delete root_node_;
    }
```

Comment (SergioRAgostinho) [reply octree/include/pcl/octree/octree_base.h]: The null pointer check is not required.

> In all cases, if ptr is a null pointer, the standard library deallocation functions do nothing. 

[Source](https://en.cppreference.com/w/cpp/memory/new/operator_delete)

Comment (kunaltyagi) [reply octree/include/pcl/octree/octree_base.h]: 😢

Comment (taketwo) [reply octree/include/pcl/octree/octree_base.h]: Oops, sorry for merging too fast! :grimacing:

**Pull request merged**